sim/src/depends.rs

```diff
@@ -0,0 +1,140 @@
+//! Support and tests related to the dependency management for multi-image
+//! support.
+
+use crate::image::ImageVersion;
+
+pub trait Depender {
+    /// Generate a version for this particular image.  The slot indiates
+    /// which slot this is being put in.
+    fn my_version(&self, offset: usize, slot: usize) -> ImageVersion;
+
+    /// Return dependencies for this image/slot combination.
+    fn my_deps(&self, offset: usize, slot: usize) -> Vec<ImageVersion>;
+
+    /// Return the image ID of the other version.
+    fn other_id(&self) -> u8;
+}
+
+/// A boring image is used when we aren't testing dependencies.  There will
+/// be meaningful version numbers.  The size field is the image number we
+/// are.
+pub struct BoringDep(pub usize);
+
+impl Depender for BoringDep {
+    fn my_version(&self, offset: usize, slot: usize) -> ImageVersion {
+        ImageVersion {
+            major: self.0 as u8 * 20 + slot as u8,
```

Comment:
I am not sure why you are multiplying by `20` (instead of any other value) here and in many other places below, but maybe this could be a constant?

Reply:
I just wanted the version numbers to be distinct and clear between the two images. I'm not sure what I'd even call the constant, but I can name it something.

Reply:
Even better, I'll make a special constructor for ImageVersion that makes the arguments clearer as well.